File: dogpack/apps/1d/diffusion/smooth_example/plotq1.py

#----------------------------------------------------------
def plotq1(m,meth1,meqn,mx,time,xc,qsoln,auxsoln):
    
    import matplotlib.pyplot as plt
    import numpy as np

    plt.figure(1)
    plt.clf()
    plt.gca().set_aspect('auto')
    plt.gca().set_xlim([xc[0],xc[mx-1]])
    plt.gca().set_ylim([-1.1,1.1])
    plt.plot(xc,qsoln[:,m],'b-')
    tmp1 = "".join(("q(t,x) at t = ",str(time)))
    title = "".join((tmp1,"     [DoGPack]"))
    plt.title(title)
    plt.draw()
    
#----------------------------------------------------------

File: dogpack/apps/1d/advection/source_example/plotq1.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%
%%   Information you have available to you:
%%
%%     Basic information:
%%              mx:  number of points
%%    [xlow,xhigh]:  min/max values of grid
%%            meqn:  number of equations
%%            maux:  number of aux components
%%           meth1:  spatial order of accuracy
%%
%%   Grid information:
%%              xc: grid points (cell centers), size = (mx,my)
%%
%%   Solution information:
%%           qsoln:  solution sampled on mesh, size = (mx,meqn)
%%             aux:  aux components sampled on mesh, size = (mx,maux)
%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%qex = exp(10*xc);
qex = 1+sin(2*pi*xc);

figure(1);
clf;
pt=plot(xc,qex,'r-');
set(pt,'linewidth',1.5);
hold on;
pz=plot(xc,qsoln,'bo');
set(pz,'markersize',8)
set(pz,'linewidth',1);
hold off;
axis on; box on; grid off;
axis([0 1 -0.1 2.1]);
set(gca,'plotboxaspectratio',[2 1 1]);
set(gca,'xtick',-2:0.25:2);
set(gca,'ytick',0:0.5:2);
set(gca,'fontsize',16);
t1 = title(['q(t,x) at t = ',num2str(time),'     [DoGPack]']); 
set(t1,'fontsize',16);

err = norm(qsoln-qex,2)/norm(qex,2);
disp(['   dx = ',num2str(dx,'%0.8e'),'         err = ',num2str(err,'%0.8e')]);
